### PR TITLE
Update the lambda-scala tutorial

### PR DESCRIPTION
**Pull request by oilnam**
Added use cases and limitations, fixed broken links and other small improvements.

### CODE REVIEW

**Comment by piuccio** [comment]
good stuff, the best part is changing `<3` to 🙈

**Comment by oilnam** [comment]
@piuccio love is blind 😇

File: docs/write-lambdas-in-scala.md [hunk 1]
```diff
@@ -1,9 +1,32 @@
...
+
+### Lambda limitations:
+
+- Max timeout hardcoded to 5 minutes (deafult being 1 minute). It means that whatever your lambda does, it has to be done within 5 minutes top.
+- Unpredictable spin up overhead when running software on the JVM (ScalaJS anyone?).
+- No control whatsoever on the hardware (obviously). You can find a list of physical limitations [here](http://docs.aws.amazon.com/lambda/latest/dg/limits.html).
+
 ### Create a new Lambda
 
 1. First you need to package up your project. One easy way to do it is with [sbt assembly](https://github.com/sbt/sbt-assembly).
```
Comment (adamnfish) [inline docs/write-lambdas-in-scala.md:44]: The improvements and fixes are great, but it's still missing bullet point "0", a guide on creating a lambda with Scala. Given this is titled `Creating a Lambda with Scala` it feels like a critical ommision.

Would it be better named "a few thoughts on whether you should use a lambda and some guidelines on deploying the lambda after you've made it"?

**Comment by oilnam** [comment]
@adamnfish good point, I added a section for it.

**Comment by adamnfish** [comment]
:+1:

**Pull request merged**